src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -61,13 +61,13 @@ protected InProcDataCollectionExtensionManager(string? runSettings, ITestEventsP
         _inProcDataCollectionSink = new InProcDataCollectionSink();
         _defaultCodeBase = defaultCodeBase;
         _fileHelper = fileHelper;
-        _codeBasePaths = new List<string?> { _defaultCodeBase };
+        CodeBasePaths = new HashSet<string?> { _defaultCodeBase };
```

Comment:
It's not related to your change but it seems that we can pass `null` for `defaultCodeBase`, it would be nice to have a test for that.